tests/network/network.go

```diff
@@ -0,0 +1,22 @@
+package network
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Defines the interface for the network setup functions used in the E2E tests
+type Network interface {
+	GetSubnetsInfo() []utils.SubnetTestInfo
```

Comment:
Thoughts on updating this interface to have `GetSubnetsInfo` instead return individual `SubnetTestInfo` instances? 

I think having it return a variable length slice/array makes it error prone for callers to check the result length and then index into the result afterwards.

Reply:
@michaelkaplan13 Not sure exactly what you mean here? Do you mean create a copy of all the internal fields, ex: dialing a new wsclient?

Reply:
do you mean something like `GetSubnetInfoByChainID(chainID) SubnetTestInfo`?

Reply:
I was thinking `GetSubnetInfo() (utils.SubnetTestInfo, utils.SubnetTestInfo)` such that it could be used as:
```
subnetA, subnetB := network.GetSubnetInfo()
```
rather than having to worry about bounds checking, additional assignments, etc.

`GetSubnetInfoByChainID(chainID) SubnetTestInfo` works too, though the caller may not know the blockchain ID they are looking for.

Reply:
The tests assume symmetry for all subnets that they interact with, so it doesn't really matter what we call subnet A vs subnet B in each test. IMO adding an `Expect` statement that the list returned by `GetSubnetsInfo` contains at least enough subnets to run the test is sufficient. 

My 2c.

Reply:
Agree. As long as there is enough subnets for the test, it should good. Added `Expect` check after `GetSubnetsInfo`.